decent-dao/fractal-interface
# Feature import token dao

## Description
This pull request updates the 'proposals' page by adding wrap and unwrap buttons. These buttons open a modal form that enables users to perform the wrap and unwrap actions. The modals have been built, and Yup validation has been added to the form. A new custom hook, `useApproval`, has been created to handle the approval steps.

## Notes
- Added wrap and unwrap buttons to the 'proposals' page
- Implemented modal forms for wrap and unwrap actions
- Added Yup validation to the form
- Created a custom hook `useApproval` for handling approval steps

## Issue / Notion doc (if applicable)

## Testing
1. Navigate to the 'proposals' page
2. Click on the wrap or unwrap button to open the respective modal form
3. Fill out the form and observe Yup validation in action
4. Submit the form and verify that the approval steps are handled by the `useApproval` custom hook

## Screenshots (if applicable)
![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop)](https://user-images.githubusercontent.com/38386583/234403209-a3099811-030d-48dc-81d9-9105c2dae91d.png)

![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop) (1)](https://user-images.githubusercontent.com/38386583/234403216-775c7ced-ae64-4669-bfff-405a391fa5b7.png)

![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop) (2)](https://user-images.githubusercontent.com/38386583/234403222-d92555e2-3b62-4e7a-8422-8baa806402ee.png)

![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop) (3)](https://user-images.githubusercontent.com/38386583/234403228-9d532495-6df4-40c1-88a7-7732ce6c9f30.png)

![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop) (4)](https://user-images.githubusercontent.com/38386583/234403238-69bef514-7c9f-47f7-8514-b2de75da768c.png)

![localhost_3000_daos_0x2300d20DA2c4571051Ed9cFf5B865A6468Ff765e_proposals(Desktop) (5)](https://user-images.githubusercontent.com/38386583/234403245-c9537307-5914-42be-9641-237aac35ff44.png)
